src/Database/Type/EnumType.php

```diff
@@ -0,0 +1,207 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Type;
+
+use BackedEnum;
+use Cake\Database\Driver;
+use Cake\Database\Exception\DatabaseException;
+use Cake\Database\TypeFactory;
+use Cake\Utility\Text;
+use InvalidArgumentException;
+use PDO;
+use ReflectionEnum;
+use ReflectionException;
+
+/**
+ * Enum type converter.
+ *
+ * Use to convert string data between PHP and the database types.
+ */
+class EnumType extends BaseType
+{
+    /**
+     * The type of the enum which is either string or int
+     *
+     * @var string
+     */
+    protected string $backingType;
+
+    /**
+     * The enum classname which is associated to the type instance
+     *
+     * @var string
+     */
+    protected string $enumClassName;
+
+    /**
+     * @param string $name The name identifying this type
+     * @param string $enumClassName The associated enum class name
+     */
+    public function __construct(
+        string $name,
+        string $enumClassName
+    ) {
+        parent::__construct($name);
+        $this->enumClassName = $enumClassName;
+        try {
+            $reflectionEnum = new ReflectionEnum($enumClassName);
+            $this->backingType = (string)$reflectionEnum->getBackingType();
+        } catch (ReflectionException) {
+            throw new DatabaseException(
+                sprintf('Unable to map enum %s for type %s, must be a backed enum.', $enumClassName, $name)
+            );
+        }
```

Comment:
```suggestion
        } catch (ReflectionException $e) {
            throw new DatabaseException(
                sprintf('Unable to map enum %s for type %s, must be a backed enum.', $enumClassName, $name),
                null,
                $e
            );
        }
```

Having the wrapped exception might help developer figure out what the problem is more easily. :shrug: